internal/provider/cluster_data_source.go

```diff
@@ -362,39 +365,51 @@ func populateClusterDataSource(
 
 	// Populate LoginNodes
 	if loginNodes, ok := cluster.GetLoginNodesOk(); ok {
-		loginNodesMap, err := loginNodes.ToMap()
-		if err != nil {
-			resp.Diagnostics.AddError(
-				"Error occured while retrieving cluster list",
-				fmt.Sprintf("Error: %v", err),
-			)
-		}
-		delete(loginNodesMap, "healthyNodes")
-		delete(loginNodesMap, "unhealthyNodes")
-
-		loginNodesObjectMap, diags := types.MapValueFrom(
-			ctx,
-			types.StringType,
-			loginNodesMap,
-		)
-		loginNodesObjectMapElements := loginNodesObjectMap.Elements()
-
-		loginNodesObjectMapElements["healthyNodes"] = types.NumberValue(
-			big.NewFloat(float64(loginNodes.GetHealthyNodes())),
-		)
-		loginNodesObjectMapElements["unhealthyNodes"] = types.NumberValue(
-			big.NewFloat(float64(loginNodes.GetUnhealthyNodes())),
-		)
-
-		resp.Diagnostics.Append(diags...)
-		loginNodesObject, diags := types.ObjectValue(
-			loginNodesObjectTypes,
-			loginNodesObjectMapElements,
-		)
-		resp.Diagnostics.Append(diags...)
-		tfClusterMapElements["loginNodes"] = loginNodesObject
+	    loginNodesList := make([]types.Object, 0)
+	    for _, loginNode := range loginNodes{
+	        loginNodeMap, err := loginNode.ToMap()
+            if err != nil {
+                resp.Diagnostics.AddError(
+                    "Error occured while retrieving cluster list",
```

Comment:
Error message must be fixed because we are retrieving the list of login nodes pools here.
This mistake exists before your PR, let's take the chance to fix it.